client/src/test/extension.test.ts

```diff
@@ -53,4 +32,25 @@ describe('Extension Tests', () => {
             expect(languages.indexOf('yml')).not.toBe(-1);
         });
     });
+    it('should get text edits from document format feature', (done) => {
+        const fileUri = vscode.workspace.textDocuments[0].uri;
+        expect(fileUri.path.endsWith('documentFormat.dcl')).toBeTruthy();
+        formatDocument(fileUri)
+            .then((list) => {
+                expect(list).toBeTruthy();
+                expect(list.length).toBe(8);
+            })
```

Comment:
Only positions? No “insert _that_ text instead”, etc.?

If you can only rely on the cardinality of the change array, maybe it would be better to have concise examples instead of huge ones. When expecting 297529 changes, you end up having no idea where they come from; when expecting just a single change in a tiny example, it’s easier to avoid false negatives. But of course the best thing would probably be genuine “before vs. after vs. expectations” tests, with a full comparison to see if the formatter spouted exactly the document we expected.

Reply:
You're right. It may also increase the code coverage and test quality.